pkg/operator/deploy/deploy.go

```diff
@@ -298,11 +298,11 @@ func (o *operator) CreateOrUpdate(ctx context.Context) error {
 }
 
 func (o *operator) IsReady(ctx context.Context) (bool, error) {
-	ok, err := ready.CheckDeploymentIsReady(ctx, o.kubernetescli.AppsV1().Deployments(pkgoperator.Namespace), "aro-operator-master")()
+	ok, err := ready.CheckAROOperatorDeploymentIsReady(ctx, o.kubernetescli.AppsV1().Deployments(pkgoperator.Namespace), "aro-operator-master")
```

Comment:
I think this is misleading that `IsReady` returns `false` when version is not a match. Also nothing in this function indicates that we check version.

Maybe we need add an additional condition similar to [`aroDeploymentReady`](https://github.com/Azure/ARO-RP/blob/339345cbb4ea1521545560f9c6e6f919ff46185f/pkg/cluster/arooperator.go#L25-L31) to check version on update? I guess it will be used somewhere [here](https://github.com/Azure/ARO-RP/blob/339345cbb4ea1521545560f9c6e6f919ff46185f/pkg/cluster/install.go#L88).

Or rename & modify `aroDeploymentReady` to do that (note that we also use it in [`Install`](https://github.com/Azure/ARO-RP/blob/339345cbb4ea1521545560f9c6e6f919ff46185f/pkg/cluster/install.go#L177)).